File: credit.c
#include <cs50.h>
#include <stdio.h>


int main(void)
{
    int numDigits;
    long num;
    int i=0;
    int creditnum[16];
    long Numorg = get_long_long("Number: ");
    do
    {
        i=0;
        numDigits = 0;
        num = Numorg;
        while (num != 0) {
            creditnum[i] = num % 10;
            num = num / 10;
            numDigits++;
            i += 1;
        }
        if(numDigits<13 || numDigits>16){
            printf("INVALID\n");
        }
    }while(numDigits<13 || numDigits>16);
    int firstDigit = creditnum[numDigits-1];
    int secDigit = creditnum[numDigits-2];
    int sum=0;
    int a;
    int b;
    for(i = 1; i<numDigits;i+=2){
        creditnum[i] = creditnum[i]*2;
        if(creditnum[i]>9){
            a = creditnum[i] % 10;
            creditnum[i] /= 10;
            b = creditnum[i] % 10;
            sum+=a+b;
        }else{
            sum+=creditnum[i];
        }
    }
    for( i = 0; i<numDigits;i+=2){
        sum+=creditnum[i];
    }
    //bool isValid = sum%10 == 0;
    if (firstDigit == 4 && sum%10 == 0 && (numDigits == 13 || numDigits == 16))
    {
        printf("VISA\n");
    }
    else if (firstDigit == 3 && (secDigit == 7 || secDigit == 4) && sum%10 == 0 && numDigits == 15)
    {
        printf("AMEX\n");
    }
    else if (firstDigit == 5 && (secDigit >= 1 && secDigit <= 5) && sum%10 == 0 && numDigits == 16)
    {
        printf("MASTERCARD\n");
    }
    else{printf("INVALID\n");}
}

File: credit.py
i=0
creditnum=[]
Numorg = int(input("Number: "))
while(numDigits<13 or numDigits>16):
    i=0
    numDigits = 0
    num = Numorg
    while (num != 0):
        creditnum.append(num % 10)
        num = num // 10
        numDigits += 1
        i += 1
    if(numDigits<13 or numDigits>16):
        break
firstDigit = creditnum[numDigits-1]
secDigit = creditnum[numDigits-2]
sum=0
a=0
b=0
for i in range(1,numDigits,2):
    creditnum[i] = creditnum[i]*2
    if(creditnum[i]>9):
        a = creditnum[i] % 10
        creditnum[i] //= 10
        b = creditnum[i] % 10
        sum=sum+a+b
    else:
        sum+=creditnum[i]
for i in range(0,numDigits,2):
    sum+=creditnum[i]
if (firstDigit == 4 and sum%10 == 0 and (numDigits == 13 or numDigits == 16)):
    print("VISA")
elif (firstDigit == 3 and (secDigit == 7 or secDigit == 4) and sum%10 == 0 and numDigits == 15):
    print("AMEX")
elif (firstDigit == 5 and (secDigit >= 1 and secDigit <= 5) and sum%10 == 0 and numDigits == 16):
    print("MASTERCARD")
else:
    print("INVALID")
